app_interface.rb

```diff
@@ -0,0 +1,40 @@
+require_relative 'app'
+
+class AppInterface
+  ACTIONS = {
+    1 => :list_books,
+    2 => :list_people,
+    3 => :create_person,
+    4 => :create_book,
+    5 => :create_rental,
+    6 => :list_rentals_for_person,
```

Comment:
- [ ] I would recommend adding all requests for input and input validation in the app_inteface.rb file and/or classes/methods whose responsibilities are to request or validate input.

![Screenshot from 2023-11-15 01-07-50](https://github.com/maximoortelli/OOP_School_Library/assets/26767383/e5e5b353-5fe0-4678-aa92-d8c5a826232c)

As in the above screenshot, you have asked the input in the app.rb. Instead you can create another method like `create_person_option` in the app_interface.rb, then take input here and call `create_person` method where logic written.

Hint:
```
def create_person_option
    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
    person_type = gets.chomp
    @app.create_person(person_type)
     ...other code
  end
```

- [ ] Same you can do while creating books as well.

Reply:
Done, thank you 👍😃